src/App.js

```diff
@@ -1,20 +1,32 @@
-import * as React from "react";
+import React, { useState, useEffect } from "react";
 import TodoList from "./TodoList.js";
 import AddTodoForm from "./AddTodoForm.js";
 
+function useSemiPersistentState() {
+  const [todoList, setTodoList] = useState(
+    JSON.parse(localStorage.getItem("savedTodoList"))
+  );
```

Comment:
when we first run the app todoList is returned as null bc there is no savedTodoList, this cause map function is failing, we can prevent this setting todoList as empty array when it savedTodoList is null

```
  const [todoList, setTodoList] = useState(
    JSON.parse(localStorage.getItem("savedTodoList")) || []
  );
```